internal/gps/deduce.go

```diff
@@ -588,6 +590,8 @@ func (dc *deductionCoordinator) deduceRootPath(ctx context.Context, path string)
 		switch d := data.(type) {
 		case maybeSource:
 			return pathDeduction{root: prefix, mb: d}, nil
```

Comment:
this is probably a much better place to think about hooking in: instead of messing with the deduction process itself, which is really not where the registry belongs (per other comments), the thing to be taking over is the `maybeSource`s returned in the `pathDeduction`s. create a new type (the `maybeRegistrySource` in the current iteration of the PR isn't structured for this at all) that takes _other_ `maybeSource`s, and figures out how to transparently wrap them.

side note with that - it'll be very important that the path on disk used by registry-wrapped ones are entirely different from what would've been used by the underlying wrapped instance.